src/rollback_to_stable/rts_btree.c

```diff
@@ -577,11 +577,11 @@ __rts_btree_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_REF *ref, WT_ROW *rip,
 
```

Comment:
@korteland I can't link to the exact line but just above on line 573 we're calling one of `__rts_btree_row_modify` or `__rts_btree_col_modify` even though inside the functions they won't insert the update due to an `if(!dryrun)` check.

I'm considering wrapping these function calls inside their own `if(!dryrun)` branch to make it more obvious that `__rts_btree_ondisk_fixup_key` never modifies the DB in dryrun mode with:
```
if(!dryrun) {
    if (rip != NULL)
        WT_ERR(__rts_btree_row_modify(session, ref, upd, key));
    else
        WT_ERR(__rts_btree_col_modify(session, ref, upd, recno));
}
```

Running our rollback_to_stable python tests indicates no problem with this change. Are there any side effects I'm missing here? The only notable change is we would no longer call `__wt_row_search`/`__wt_col_search` inside of our `*_modify` functions which may impact some statistocs.

Reply:
I'd prefer to leave it as-is, the search could fail which will bubble back up the call stack and potentially be a useful signal that RTS did something wrong. As a rule of thumb, dry-run should be as close as reasonably possible to the real thing. I'm happy to discuss this further if you feel really strongly about it though.

Reply:
Nah no worries. Just saw a possible tweak we could make. Resolving comment